### DISCUSSION TITLE
DigitalOcean Failed to Deploy

### DISCUSSION THREAD

**Discussion by shaneoliver** [directus/directus#16548] [Archive]
I'm having an issue deploying an app on DigitalOcean. The database connection is fine as it migrates the tables and creates the admin user but the deployment always fails with the following

 Deploy Error: Non-Zero Exit Code

![Screen Shot 2022-11-21 at 10 57 48 am](https://user-images.githubusercontent.com/18100929/202933740-30b52c0e-f803-432e-b08f-bf9894667841.png)

The error in DigitalOcean isn't that helpful and there is nothing in the logs to help debug this
https://docs.digitalocean.com/products/app-platform/reference/error-codes/#deploy-errors

**Comment by azrikahar** [comment]
You might be facing the same issue in #16382 in regards with the incompatible Node version. You can try the solution shared over here: https://github.com/directus/directus/discussions/16382#discussioncomment-4178051, but it's also strongly recommended to use the docker container in DO App Platform going forward, as the 1-click install isn't actively maintained as mentioned here: https://github.com/directus/directus/discussions/16382#discussioncomment-4171582.

Note: just in case it may not be the same error, do try to set the env variable `LOG_STYLE` to `raw` to check whether there's any error logs that weren't showing up, as the there is a known issue #16284 with the default `pretty` which is not able to print the log in time when the process exits due to it's (current) async nature.

**Comment by shaneoliver** [comment]
Thanks for the info. To clarify for future readers, this was not via the one-click installer it was a managed database and app platform. I added LOG_STYLE=raw (great tip) and redeployed the app after several hours and it worked. This app was created on the new SYD1 data centre,  I imagine the deployment failing was an issue on DigitalOcean's end which is now resolved. I had previously tried the docker container on app platform with the deployment also failing
